{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c94d6ba4-609a-4cd0-b33f-47eb038db462",
   "metadata": {},
   "source": [
    "# Clinical data cleaning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ac7063a9-03d4-4506-a3a1-4b553e7ca548",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the already-filtered file\n",
    "file_path = r\"C:\\Users\\DELL\\Downloads\\clinvar_result.txt\"\n",
    "df = pd.read_csv(file_path, sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "62091a6c-518b-4b39-a2d0-19d5a840ec82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene(s)</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>Condition(s)</th>\n",
       "      <th>Accession</th>\n",
       "      <th>GRCh37Chromosome</th>\n",
       "      <th>GRCh37Location</th>\n",
       "      <th>GRCh38Chromosome</th>\n",
       "      <th>GRCh38Location</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>...</th>\n",
       "      <th>Germline classification</th>\n",
       "      <th>Germline date last evaluated</th>\n",
       "      <th>Germline review status</th>\n",
       "      <th>Somatic clinical impact</th>\n",
       "      <th>Somatic clinical impact date last evaluated</th>\n",
       "      <th>Somatic clinical impact review status</th>\n",
       "      <th>Oncogenicity classification</th>\n",
       "      <th>Oncogenicity date last evaluated</th>\n",
       "      <th>Oncogenicity review status</th>\n",
       "      <th>Unnamed: 24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_000492.4(CFTR):c.1A&gt;G (p.Met1Val)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>M1V</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>VCV000053423</td>\n",
       "      <td>7</td>\n",
       "      <td>117120149</td>\n",
       "      <td>7</td>\n",
       "      <td>117480095</td>\n",
       "      <td>53423</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Mar 17, 2017</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_000492.4(CFTR):c.4C&gt;T (p.Gln2Ter)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>Q2*</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>VCV000053980</td>\n",
       "      <td>7</td>\n",
       "      <td>117120152</td>\n",
       "      <td>7</td>\n",
       "      <td>117480098</td>\n",
       "      <td>53980</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Mar 17, 2017</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NM_000492.4(CFTR):c.11C&gt;A (p.Ser4Ter)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>S4*</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>VCV000053211</td>\n",
       "      <td>7</td>\n",
       "      <td>117120159</td>\n",
       "      <td>7</td>\n",
       "      <td>117480105</td>\n",
       "      <td>53211</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Mar 17, 2017</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_000492.4(CFTR):c.38C&gt;T (p.Ser13Phe)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>S13F</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>VCV000053845</td>\n",
       "      <td>7</td>\n",
       "      <td>117120186</td>\n",
       "      <td>7</td>\n",
       "      <td>117480132</td>\n",
       "      <td>53845</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Aug 31, 2018</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_000492.4(CFTR):c.44T&gt;C (p.Leu15Pro)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>L15P</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>VCV000634833</td>\n",
       "      <td>7</td>\n",
       "      <td>117120192</td>\n",
       "      <td>7</td>\n",
       "      <td>117480138</td>\n",
       "      <td>634833</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Aug 31, 2018</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Name Gene(s) Protein change  \\\n",
       "0    NM_000492.4(CFTR):c.1A>G (p.Met1Val)    CFTR            M1V   \n",
       "1    NM_000492.4(CFTR):c.4C>T (p.Gln2Ter)    CFTR            Q2*   \n",
       "2   NM_000492.4(CFTR):c.11C>A (p.Ser4Ter)    CFTR            S4*   \n",
       "3  NM_000492.4(CFTR):c.38C>T (p.Ser13Phe)    CFTR           S13F   \n",
       "4  NM_000492.4(CFTR):c.44T>C (p.Leu15Pro)    CFTR           L15P   \n",
       "\n",
       "      Condition(s)     Accession  GRCh37Chromosome GRCh37Location  \\\n",
       "0  Cystic fibrosis  VCV000053423                 7      117120149   \n",
       "1  Cystic fibrosis  VCV000053980                 7      117120152   \n",
       "2  Cystic fibrosis  VCV000053211                 7      117120159   \n",
       "3  Cystic fibrosis  VCV000053845                 7      117120186   \n",
       "4  Cystic fibrosis  VCV000634833                 7      117120192   \n",
       "\n",
       "   GRCh38Chromosome GRCh38Location  VariationID  ...  Germline classification  \\\n",
       "0                 7      117480095        53423  ...               Pathogenic   \n",
       "1                 7      117480098        53980  ...               Pathogenic   \n",
       "2                 7      117480105        53211  ...               Pathogenic   \n",
       "3                 7      117480132        53845  ...               Pathogenic   \n",
       "4                 7      117480138       634833  ...               Pathogenic   \n",
       "\n",
       "  Germline date last evaluated    Germline review status  \\\n",
       "0                 Mar 17, 2017  reviewed by expert panel   \n",
       "1                 Mar 17, 2017  reviewed by expert panel   \n",
       "2                 Mar 17, 2017  reviewed by expert panel   \n",
       "3                 Aug 31, 2018  reviewed by expert panel   \n",
       "4                 Aug 31, 2018  reviewed by expert panel   \n",
       "\n",
       "  Somatic clinical impact Somatic clinical impact date last evaluated  \\\n",
       "0                     NaN                                         NaN   \n",
       "1                     NaN                                         NaN   \n",
       "2                     NaN                                         NaN   \n",
       "3                     NaN                                         NaN   \n",
       "4                     NaN                                         NaN   \n",
       "\n",
       "  Somatic clinical impact review status Oncogenicity classification  \\\n",
       "0                                   NaN                         NaN   \n",
       "1                                   NaN                         NaN   \n",
       "2                                   NaN                         NaN   \n",
       "3                                   NaN                         NaN   \n",
       "4                                   NaN                         NaN   \n",
       "\n",
       "  Oncogenicity date last evaluated  Oncogenicity review status  Unnamed: 24  \n",
       "0                              NaN                         NaN          NaN  \n",
       "1                              NaN                         NaN          NaN  \n",
       "2                              NaN                         NaN          NaN  \n",
       "3                              NaN                         NaN          NaN  \n",
       "4                              NaN                         NaN          NaN  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "79f68656-abb9-439b-a5a8-cd0733e713ff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top Genes:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Gene(s)\n",
       "CFTR    226\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Step 1: Confirm CFTR is present\n",
    "print(\"Top Genes:\")\n",
    "df['Gene(s)'].value_counts().head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4519567d-e702-4bbc-9db6-5a013ff26060",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Step 2: Filter for CFTR gene\n",
    "df_cftr = df[df['Gene(s)'] == 'CFTR']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2463a77b-fa03-4e1b-9c82-082a3d68493d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Top conditions for CFTR:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Condition(s)\n",
       "Cystic fibrosis                                  214\n",
       "ivacaftor response - Efficacy|Cystic fibrosis      6\n",
       "Cystic fibrosis|ivacaftor response - Efficacy      6\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Step 3: See what conditions exist\n",
    "print(\" Top conditions for CFTR:\")\n",
    "df_cftr['Condition(s)'].value_counts().head(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a9b4ed7e-ddd7-4654-9b0e-862c6d35ada1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Just keep useful columns\n",
    "columns_to_keep = [\n",
    "    'Name',\n",
    "    'Gene(s)',\n",
    "    'Protein change',\n",
    "    'Variant type',\n",
    "    'Molecular consequence',\n",
    "    'Germline classification',\n",
    "    'Germline review status',\n",
    "    'GRCh38Chromosome',\n",
    "    'GRCh38Location'\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "52db583f-dbb8-424b-9f54-d5fecbb525cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Clean data\n",
    "df_clean = df_cftr[columns_to_keep].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4e716060-32a0-41f4-823e-27c0c50193b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Cleaned file saved. Preview:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene(s)</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>Variant type</th>\n",
       "      <th>Molecular consequence</th>\n",
       "      <th>Germline classification</th>\n",
       "      <th>Germline review status</th>\n",
       "      <th>GRCh38Chromosome</th>\n",
       "      <th>GRCh38Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>NM_000492.4(CFTR):c.4144C&gt;T (p.Gln1382Ter)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>Q1382*</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>7</td>\n",
       "      <td>117665466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>NM_000492.4(CFTR):c.4231C&gt;T (p.Gln1411Ter)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>Q1411*</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>7</td>\n",
       "      <td>117665553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>NM_000492.4(CFTR):c.4234C&gt;T (p.Gln1412Ter)</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>Q1412*</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>7</td>\n",
       "      <td>117665556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>NM_000492.4(CFTR):c.4242+1G&gt;T</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>7</td>\n",
       "      <td>117665565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>NM_000492.4(CFTR):c.4242+1G&gt;A</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "      <td>7</td>\n",
       "      <td>117665565</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Name Gene(s) Protein change  \\\n",
       "221  NM_000492.4(CFTR):c.4144C>T (p.Gln1382Ter)    CFTR         Q1382*   \n",
       "222  NM_000492.4(CFTR):c.4231C>T (p.Gln1411Ter)    CFTR         Q1411*   \n",
       "223  NM_000492.4(CFTR):c.4234C>T (p.Gln1412Ter)    CFTR         Q1412*   \n",
       "224               NM_000492.4(CFTR):c.4242+1G>T    CFTR            NaN   \n",
       "225               NM_000492.4(CFTR):c.4242+1G>A    CFTR            NaN   \n",
       "\n",
       "                  Variant type Molecular consequence Germline classification  \\\n",
       "221  single nucleotide variant              nonsense              Pathogenic   \n",
       "222  single nucleotide variant              nonsense              Pathogenic   \n",
       "223  single nucleotide variant              nonsense              Pathogenic   \n",
       "224  single nucleotide variant  splice donor variant              Pathogenic   \n",
       "225  single nucleotide variant  splice donor variant              Pathogenic   \n",
       "\n",
       "       Germline review status  GRCh38Chromosome GRCh38Location  \n",
       "221  reviewed by expert panel                 7      117665466  \n",
       "222  reviewed by expert panel                 7      117665553  \n",
       "223  reviewed by expert panel                 7      117665556  \n",
       "224  reviewed by expert panel                 7      117665565  \n",
       "225  reviewed by expert panel                 7      117665565  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save clean file\n",
    "df_clean.to_csv(\"cftr_mutations_clean.csv\", index=False)\n",
    "print(\" Cleaned file saved. Preview:\")\n",
    "df_clean.tail()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a2972ee-555d-450a-a15b-36829e28b863",
   "metadata": {},
   "source": [
    "#  Mutation Statistics Section"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "fd58341f-87c5-4aaf-8d5a-53d82d7374b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " Mutation Type Summary:\n",
      "Variant type\n",
      "single nucleotide variant    154\n",
      "Deletion                      48\n",
      "Duplication                   18\n",
      "Indel                          4\n",
      "Insertion                      2\n",
      "Name: count, dtype: int64\n",
      "\n",
      " Top Molecular Consequences:\n",
      "Molecular consequence\n",
      "nonsense                                    69\n",
      "missense variant                            67\n",
      "frameshift variant                          63\n",
      "splice donor variant                        16\n",
      "splice acceptor variant                     10\n",
      "missense variant|initiator_codon_variant     1\n",
      "Name: count, dtype: int64\n",
      "\n",
      " Top Protein Changes:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein change\n",
       "W1204*    2\n",
       "Y913*     2\n",
       "Q685fs    2\n",
       "G628R     2\n",
       "W401*     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Mutation summary\n",
    "print(\"\\n Mutation Type Summary:\")\n",
    "print(df_clean['Variant type'].value_counts())\n",
    "\n",
    "print(\"\\n Top Molecular Consequences:\")\n",
    "print(df_clean['Molecular consequence'].value_counts())\n",
    "\n",
    "print(\"\\n Top Protein Changes:\")\n",
    "df_clean['Protein change'].value_counts().head(5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f614b46c-1066-4cbe-bc3c-805dfcb953a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant type</th>\n",
       "      <th>Molecular consequence</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>A959fs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>D1201fs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>D1202fs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>D648fs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>F143fs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>W882*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Y122*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Y275*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Y849*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Y913*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>182 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Variant type Molecular consequence Protein change  Count\n",
       "0                     Deletion    frameshift variant         A959fs      1\n",
       "1                     Deletion    frameshift variant        D1201fs      1\n",
       "2                     Deletion    frameshift variant        D1202fs      1\n",
       "3                     Deletion    frameshift variant         D648fs      1\n",
       "4                     Deletion    frameshift variant         F143fs      1\n",
       "..                         ...                   ...            ...    ...\n",
       "177  single nucleotide variant              nonsense          W882*      1\n",
       "178  single nucleotide variant              nonsense          Y122*      1\n",
       "179  single nucleotide variant              nonsense          Y275*      1\n",
       "180  single nucleotide variant              nonsense          Y849*      1\n",
       "181  single nucleotide variant              nonsense          Y913*      1\n",
       "\n",
       "[182 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary = df_clean.groupby(['Variant type', 'Molecular consequence','Protein change']).size().reset_index(name='Count')\n",
    "summary.to_csv(\"cftr_mutation_summary.csv\", index=False)\n",
    "summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "93edbbda-5abd-4ffd-8fff-007b955197d4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique mutation residue positions in CFTR: [1, 2, 4, 13, 15, 17, 19, 27, 30, 39, 46, 67, 75, 79, 88, 91, 92, 98, 99, 102, 104, 109, 110, 116, 117, 122, 126, 137, 139, 142, 143, 148, 161, 165, 177, 178, 183, 191, 192, 193, 194, 199, 205, 206, 216, 218, 220, 227, 232, 254, 268, 275, 276, 284, 287, 311, 330, 334, 336, 338, 341, 346, 347, 352, 361, 378, 379, 401, 560, 561, 563, 568, 569, 574, 585, 598, 601, 609, 613, 628, 637, 641, 648, 656, 663, 664, 673, 684, 685, 709, 710, 715, 720, 726, 732, 739, 745, 748, 764, 785, 792, 818, 821, 822, 831, 835, 846, 849, 851, 861, 864, 868, 882, 890, 912, 913, 922, 927, 938, 941, 942, 945, 953, 959, 966, 970, 979, 1128, 1145, 1158, 1159, 1179, 1201, 1202, 1204, 1231, 1234, 1240, 1244, 1249, 1250, 1251, 1254, 1255, 1258, 1269, 1274, 1283, 1295, 1297, 1298, 1301, 1303, 1310, 1313, 1324, 1330, 1335, 1349, 1360, 1363, 1366, 1371, 1375, 1376, 1382, 1383, 1411, 1412] ...\n"
     ]
    }
   ],
   "source": [
    "# Load cleaned CSV\n",
    "df = pd.read_csv(\"cftr_mutations_clean.csv\")\n",
    "\n",
    "# Extract positions from 'Protein change' (like p.Phe508del → 508)\n",
    "df['Position'] = df['Protein change'].str.extract(r'(\\d+)')\n",
    "\n",
    "# Drop null positions\n",
    "df_pos = df.dropna(subset=['Position'])\n",
    "\n",
    "# Save list of positions to use in Chimera\n",
    "positions = df_pos['Position'].unique().astype(int).tolist()\n",
    "positions.sort()\n",
    "\n",
    "print(\"Unique mutation residue positions in CFTR:\", positions[:], \"...\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c7be6f5c-9209-43f6-9c1e-401ce2aa8d05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>Germline classification</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>102</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>104</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>109</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>110</td>\n",
       "      <td>Pathogenic; drug response</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>966</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>970</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>979</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>98</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>99</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>172 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Position    Germline classification  Count\n",
       "0          1                 Pathogenic      1\n",
       "1        102                 Pathogenic      1\n",
       "2        104                 Pathogenic      1\n",
       "3        109                 Pathogenic      1\n",
       "4        110  Pathogenic; drug response      1\n",
       "..       ...                        ...    ...\n",
       "167      966                 Pathogenic      1\n",
       "168      970                 Pathogenic      2\n",
       "169      979                 Pathogenic      1\n",
       "170       98                 Pathogenic      2\n",
       "171       99                 Pathogenic      1\n",
       "\n",
       "[172 rows x 3 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group by position and classification\n",
    "\n",
    "summary = df.groupby(['Position', 'Germline classification']).size().reset_index(name='Count')\n",
    "summary.to_csv(\"cftr_classification_by_position.csv\", index=False)\n",
    "summary"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
